dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaIncomingConnector.java

```diff
@@ -135,4 +138,25 @@ private void shutdown() {
         }
     }
 
+    private <K, V> KafkaConsumer<K, V> getKafkaConsumerWithRetry(Map<String, Object> consumerConfig, int retrySeconds, String channelName) throws InterruptedException {
+        if (retrySeconds == 0) {
+            return this.kafkaAdapterFactory.newKafkaConsumer(consumerConfig);
+        }
+
+        long retryNs = Duration.ofSeconds(retrySeconds).toNanos();
+        long startTime = System.nanoTime();
+
+        while (true) {
+            try {
+                return this.kafkaAdapterFactory.newKafkaConsumer(consumerConfig);
```

Comment:
Can we sync up with SmallRye for the `retry-max-wait`, `retry-attempts`?

Reply:
I'm not sure what the point of those options are in smallrye, they don't seem to be the same as this one.

They talk about retrying the connection if it goes down, which is not what this change is doing.

Reply:
I named the property `creation.retry.seconds` to emphasise that it's only the _creation_ of the kafka client which is retried.

Once we have set up the kafka client, we leave it to do any retries itself.